presto-main/src/main/java/com/facebook/presto/operator/aggregation/MergeTDigestFunction.java

```diff
@@ -0,0 +1,167 @@
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "Merges the input t digests into a single t digest";
+    }
+
+    @Override
+    public InternalAggregationFunction specialize(BoundVariables boundVariables, int arity, TypeManager typeManager, FunctionManager functionManager)
+    {
+        Type valueType = boundVariables.getTypeVariable("T");
+        TDigestType outputType = (TDigestType) typeManager.getParameterizedType(StandardTypes.TDIGEST,
+                ImmutableList.of(TypeSignatureParameter.of(valueType.getTypeSignature())));
+        return generateAggregation(outputType);
+    }
+
+    private static InternalAggregationFunction generateAggregation(TDigestType type)
+    {
+        DynamicClassLoader classLoader = new DynamicClassLoader(MapAggregationFunction.class.getClassLoader());
+        TDigestStateSerializer stateSerializer = new TDigestStateSerializer();
+        Type intermediateType = stateSerializer.getSerializedType();
+
+        AggregationMetadata metadata = new AggregationMetadata(
+                generateAggregationName(NAME, type.getTypeSignature(), ImmutableList.of(type.getTypeSignature())),
+                createInputParameterMetadata(type),
+                INPUT_FUNCTION.bindTo(type),
+                COMBINE_FUNCTION,
+                OUTPUT_FUNCTION.bindTo(stateSerializer),
+                ImmutableList.of(new AccumulatorStateDescriptor(
+                        TDigestState.class,
+                        stateSerializer,
+                        new TDigestStateFactory())),
+                type);
+
+        GenericAccumulatorFactoryBinder factory = AccumulatorCompiler.generateAccumulatorFactoryBinder(metadata, classLoader);
+        return new InternalAggregationFunction(NAME, ImmutableList.of(type), ImmutableList.of(intermediateType), type, true, true, factory);
+    }
+
+    private static List<ParameterMetadata> createInputParameterMetadata(Type valueType)
+    {
+        return ImmutableList.of(
+                new ParameterMetadata(STATE),
+                new ParameterMetadata(BLOCK_INPUT_CHANNEL, valueType),
+                new ParameterMetadata(BLOCK_INDEX));
+    }
+
+    @InputFunction
+    public static void input(Type type, TDigestState state, Block value, int index)
+    {
+        merge(state, new TDigest(type.getSlice(value, index)));
+    }
+
+    @CombineFunction
+    public static void combine(TDigestState state, TDigestState otherState)
+    {
+        merge(state, otherState.getTDigest());
+    }
+
+    private static void merge(TDigestState state, TDigest input)
+    {
+        if (input == null) {
+            return;
+        }
+        TDigest previous = state.getTDigest();
+        if (previous == null) {
+            state.setTDigest(input);
+            state.addMemoryUsage(input.estimatedSerializedSizeInBytes());
+        }
+        else {
+            checkArgument(previous.getCompressionFactor() == input.getCompressionFactor(),
```

Comment:
This should throw a PrestoException, otherwise it will appear as an internal error in Presto.